src/components/Staking/WithdrawalCredentials.tsx

```diff
@@ -0,0 +1,169 @@
+import Emoji from "../Emoji"
+import Link from "../Link"
+import Translation from "../../components/Translation"
+
+interface Validator {
+  validatorIndex: number
+  withdrawalCredentials: string
+  isUpgraded: boolean
+  isTestnet?: boolean
+}
+
+interface IProps {}
+const WithdrawalCredentials: FC<IProps> = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [hasError, setHasError] = useState<boolean>(false)
+  const [inputValue, setInputValue] = useState<string>("")
+  const [validator, setValidator] = useState<Validator | null>(null)
+  const [isTestnet, setIsTestnet] = useState<boolean>(false)
+
+  const checkWithdrawalCredentials = async () => {
+    setHasError(false)
+    setIsLoading(true)
+    try {
+      const response = await fetch(
+        `https://${
+          isTestnet ? "goerli." : ""
+        }beaconcha.in/api/v1/validator/${inputValue}`
+      )
+      const { data } = await response.json()
+      const withdrawalCredentials = data.withdrawalcredentials
+      setValidator({
+        validatorIndex: parseInt(inputValue),
+        withdrawalCredentials,
+        isUpgraded: withdrawalCredentials.startsWith("0x01"),
+        isTestnet,
+      })
+    } catch (error) {
+      console.error(error)
+      setHasError(true)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setInputValue(e.target.value.replace(/\D/g, ""))
+
+  const handleNetworkToggle = () => setIsTestnet((prev) => !prev)
+
+  const longAddress = useMemo<string>(
+    () => (validator ? `0x${validator.withdrawalCredentials.slice(-40)}` : ""),
+    [validator]
+  )
+  const shortAddress = useMemo<string>(
+    () =>
+      longAddress ? `${longAddress.slice(0, 6)}…${longAddress.slice(-4)}` : "",
+    [longAddress]
+  )
+  const resultText = useMemo<string | JSX.Element>(() => {
+    if (isLoading) return <Spinner />
+    if (hasError)
+      return "Error fetching. Double check validator index number and try again."
+    if (!validator) return " "
+    if (validator.isUpgraded)
+      return (
+        <>
+          <Text display={["none", null, "block"]}>
+            Withdrawal address:{" "}
+            <Text as="span" title={longAddress} fontWeight="bold">
+              {longAddress}
+            </Text>
+          </Text>
+          <Text>
+            <Emoji text=":tada:" mr={1} />
+            Congrats! This validator has been upgraded and will automatically
+            receive rewards to{" "}
+            <Text as="span" title={longAddress} fontWeight="bold">
+              {shortAddress}
+            </Text>{" "}
+            when the Shanghai upgrade is complete.
+          </Text>
+        </>
+      )
+    return (
+      <Text as="span">
+        <Emoji text="⚠️" mr={1} />
+        This {validator.isTestnet ? "Goerli testnet" : ""} validator has not
+        been upgraded. Instructions on how to upgrade can be found at{" "}
```

Comment:
Maybe use withdrawal credentials set/ not set vs. upgrade/not upgraded.

Reply:
Yeah, exact term to use continues to seem debatable... a lot of the devs will refer to this as "upgrading" your keys. This text is honestly being played with and iterated on a bit in the Figma... This component is being looked at from a design angle which will influence the copy we end up going with